src/baseTrie.js

```diff
@@ -49,6 +49,38 @@ module.exports = class Trie {
     this.root = root
   }
 
+  static fromProof(proofNodes, cb, proofTrie){
+    let opStack = proofNodes.map((nodeValue) => {
+      return {type: 'put', key: ethUtil.keccak(nodeValue), value: ethUtil.toBuffer(nodeValue)}
```

Comment:
its kindof like the verification is done in this step because when you traverse the tree you already know each key is the hash of the node it's storing